### DISCUSSION TITLE
你们试过官网案例的吗？性能提升了？还不如 create-react-app

### DISCUSSION THREAD

**Discussion by zhangwilling** [umijs/umi#9676] [Q&A] [answered]
使用 antd-pro 的案例，包工具为 yarn ，umi 为 latest 

1. 热更新要 5s ，最简单的案例都要5秒

2. 在 pages 下面新增文件，无法自动生成对应的路由

**Comment by zhangwilling** [comment]
只是简单的改几个字而已，就要5秒

<img width="504" alt="image" src="https://user-images.githubusercontent.com/20391734/199430106-85212c47-958e-4016-bec5-b7da4b6b4c56.png">


<img width="633" alt="image" src="https://user-images.githubusercontent.com/20391734/199430138-7b708fb2-723f-4979-8c4d-bd8513067946.png">

**Comment by qiutian00** [comment]
期待大佬们优化

**Comment by GOWxx** [comment]
# 测试
根据文档的方法, [创建项目](https://umijs.org/docs/tutorials/getting-started#%E5%88%9B%E5%BB%BA%E9%A1%B9%E7%9B%AE)
```bash
$ mkdir umiapp && cd umiapp
$ mkdir myapp && cd myapp
```
[启动项目](https://umijs.org/docs/tutorials/getting-started#%E5%90%AF%E5%8A%A8%E9%A1%B9%E7%9B%AE)
```bash
$ pnpm dev
```
## 问题一
热更新时间
我这里测试简单修改的字，一开始大概是200毫秒，后来大概是几十毫秒
![image](https://user-images.githubusercontent.com/12870715/199455294-22ad9963-8f29-4676-893e-dbe267b03d58.png)

## 问题二
pages 下面新增文件
我这里可以生成了对应路由
![image](https://user-images.githubusercontent.com/12870715/199455405-ecbb330f-39d8-4b59-947b-256d6ad11489.png)
![image](https://user-images.githubusercontent.com/12870715/199456263-839b3bb4-e977-48e9-ae42-cd8044dbbcc8.png)

## 想法

我也是按照「官网案例」来搞的最基本的 demo
可能需要你描述更具体的场景或复现方法

有了可复现的例子，才好确定哪里性能不太好，该怎么优化。

**Reply by zhangwilling** [reply]
使用 antd-pro 的案例，包工具为 yarn ，umi 为 latest

**Comment by fz6m** [comment]
有空的时候可以辛苦看下这个 MFSU eager 问题 @stormslowly  。

复现步骤，创建一个 simple umi 项目并加入 antd ：

```bash
  pnpm create umi demo
  cd ./demo
  pnpm add antd @umijs/plugins
```

配置：

```ts
// .umirc.ts

export default {
  npmClient: 'pnpm',
  antd: {},
  plugins: [
    '@umijs/plugins/dist/antd'
  ]
}
```

复现代码：

```ts
// src/pages/index.tsx

import { Button } from 'antd'

export default function HomePage() {
  return (
    <div>
      <h2>Yay! Welcome to umi!</h2>
      <Button type='primary'>kk333k111</Button>
    </div>
  );
}
```

复现步骤：

1. 删除 `<Button type='primary'>` 中的 `type='primary'` ，等待页面 HMR 完成，再添加回来，再等待完成，如此反复删增几次就会引发 MFSU 的 warn 警告：

    <img src='https://user-images.githubusercontent.com/59400654/199465812-873123be-7425-430a-a62b-7e3c325309a4.png' width='40%' />

2. 出现警告后，任何改动都需要 5s 时间延迟。

注：MFSU normal 、不启用 MFSU 无此问题，HMR 在 300ms 。

**Comment by stormslowly** [comment]
收到 明天改

**Comment by stormslowly** [comment, marked as answer]
新版的 mfsu  某些场景可能会劣化，（我的锅，轮询机制确实有问题，已经在改了）
可以用这个配置  `mfsu: {strategy: 'normal'} ` 这个是稳定的

**Comment by stormslowly** [comment]
@zhangwilling   可以试试 4.0.0-canary.20221104.2 升级了 webpack ，可以一定概率避免polling 失败的情况

**Reply by zhangwilling** [reply]
好的，辛苦了
